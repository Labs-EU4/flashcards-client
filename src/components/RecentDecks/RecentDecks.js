import React, {useState, useEffect} from "react";
import {Link} from "react-router-dom";
import {Card, Avatar} from "antd";
import {axiosWithAuth} from "../../utils/axios";
import styles from "./RecentDecks.module.css";

const {Meta} = Card;

const RecentDecks = props => {
  const [recentDecks, setRecentDecks] = useState([]);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    axiosWithAuth()
      .get("/decks/access/")
      .then(res => {
        setRecentDecks(res.data.data);
      })
      .catch(err => {
        console.log(err);
      });
  }, []);

  return (
    <div className={styles.container}>
      <h1 className={styles.heading} data-testid="heading">
        Recently played
      </h1>
      <div data-testid="decks">
        {recentDecks.length === 0 ? (
          <div className={styles.noDecks}>You haven't completed a session yet!</div>
        ) : (
          recentDecks.map((deck, index) => {
            return (
              <Card
                style={{width: 300}}
                actions={[<Link to={`/decks/${deck.deck_id}`}>Play</Link>]}
                loading={loading}
                className={styles.deckCard}
                key={index}
              >
                <Meta
                  avatar={<Avatar src="logo192.png" />}
                  description="This is the description"
                  cardnumber="52"
                />
              </Card>
            );
          })
        )}
      </div>
    </div>
  );
};

export default RecentDecks;
