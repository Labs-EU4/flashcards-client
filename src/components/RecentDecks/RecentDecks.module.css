.container {
  width: 100%;
  min-height: 48.5vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #f9f9f9;
  border-radius: 5px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.5);
  transition: 0.3s;
}

.heading {
  font-style: normal;
  font-weight: normal;
  font-size: 2vw;
  text-align: center;
  padding-top: 1.5rem;
}

.noDecks {
  font-size: 1.5vw;
  padding-top: 12vh;
}

.deckCard {
  margin-bottom: 10px;
}

@media only screen and (max-width: 1024px) {
  .container {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-right: 2%;
    height: auto;
    min-height: auto;
  }
  .heading {
    font-size: 3vw;
  }
  .noDecks {
    font-size: 2vw;
    padding: 4vh;
  }
}

@media only screen and (max-width: 600px) {
  .heading {
    font-size: 4vw;
  }
  .noDecks {
    font-size: 4vw;
    padding: 4vh;
  }
}
