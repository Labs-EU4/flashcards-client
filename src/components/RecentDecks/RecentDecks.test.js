import React from "react";
import * as rtl from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import RecentDecks from "./RecentDecks";

let wrapper;
beforeEach(() => {
  wrapper = rtl.render(<RecentDecks />);
});

afterEach(rtl.cleanup);

it("should take a snapshot", () => {
  const {asFragment} = rtl.render(<RecentDecks />);

  expect(asFragment(<RecentDecks />)).toMatchSnapshot();
});

describe("is the component rendering correctly", () => {
  it("renders the heading", () => {
    let queryValue = wrapper.getByTestId("heading");
    expect(queryValue).toBeInTheDocument();
  });

  it("renders the heading", () => {
    let queryValue = wrapper.getByTestId("decks");
    expect(queryValue).toBeInTheDocument();
  });
});
