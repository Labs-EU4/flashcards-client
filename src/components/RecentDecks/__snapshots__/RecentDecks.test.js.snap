// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1
      class="heading"
      data-testid="heading"
    >
      Recently played
    </h1>
    <div
      data-testid="decks"
    >
      <div
        class="noDecks"
      >
        You haven't completed a session yet!
      </div>
    </div>
  </div>
</DocumentFragment>
`;
